### DISCUSSION TITLE
[Help]: Carousel alignment issue when scrolling to the next page

### DISCUSSION THREAD

**Discussion by aygee** [davidjerleke/embla-carousel#1024] [Help] [answered]
### Summary

I'm building a carousel that shifts by the number of items visible in the viewport. All the items have the same width. To achieve this, I set 'slidesToScroll' to match the number of visible items.

However, sometimes the carousel doesn't align properly when moving to the next page. After debugging, I found that the issue seems to be with the scrollSnapList values. For example, with 24 items and a scroll of 8 items at a time, the scrollSnapList should be [0, 0.5, 1]. However, sometimes, it's returning a floating number, eg: [0, 0.5333186716516385, 1], which I think caused the misalignment.

I'm not sure what caused it. However, I wonder if we can override the scrollSnapList, because this seems to be easily calculated in this specific scenario?

Thanks.

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by aygee** [comment]
After digging a bit more, it looks like the issue only happens when I lazyload the component when it comes to view using the browser Intersection Observer API.

When I removed the lazy load, it works perfectly fine! It's very strange!

**Reply by davidjerleke** [reply]
@aygee I can't help you and debug if I don't have access to your setup in a CodeSandbox.

**Reply by aygee** [reply]
@davidjerleke I understand. I'll see if I can create a minimal reproduction. Thanks.

**Comment by aygee** [comment, marked as answer]
@davidjerleke I found the bug. I had `transition: 'transform 175ms ease'` on the embla container and this caused the issue and also some jankiness when scrolling. Removing it makes the scroll much smoother and I don't see the issue anymore!

Thanks for making such a great library!
